src/BasicRum/Diagram/Builder/RenderType/Plane.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\BasicRum\Diagram\Builder\RenderType;
+
+use App\BasicRum\Diagram\Builder\RenderType\Metrics\PlaneBusinessMetrics;
+use App\BasicRum\Diagram\Builder\RenderType\Metrics\PlaneTechnicalMetrics;
+use App\BasicRum\Diagram\View\Layout;
+use App\BasicRum\Diagram\View\RenderType\Plane as ViewRenderTypePlane;
+use App\BasicRum\DiagramOrchestrator;
+use App\BasicRum\Release;
+use App\BasicRum\RenderTypeInterface;
+
+class Plane implements RenderTypeInterface
+{
+    private $diagramOrchestrator;
+    private $params;
+    private $releaseRepository;
+    private $dataForDiagram;
+    private $extraDiagramParams;
+    private $extraLayoutParams;
+    private $results;
+
+    public function __construct(DiagramOrchestrator $diagramOrchestrator, array $params, Release $releaseRepository)
+    {
+        $this->diagramOrchestrator = $diagramOrchestrator;
+        $this->params = $params;
+        $this->releaseRepository = $releaseRepository;
+        $this->dataForDiagram = [];
+        $this->extraDiagramParams = [];
+        $this->extraLayoutParams = $this->getExtraLayoutParams($params);
+        $this->results = $diagramOrchestrator->process();
+    }
+
+    public function build(): array
+    {
+        $hasError = false;
+
+        try {
+            $businessMetrics = new PlaneBusinessMetrics($this->results, $this->params);
+            $technicalMetrics = new PlaneTechnicalMetrics($this->results, $this->params);
+            foreach ($this->results as $key => $result) {
+                $businessMetrics->proceed($key);
```

Comment:
Could you explain what proceed means in this context?

Reply:
Hello @ceckoslab 
It's to start / take action. 
According to https://www.merriam-webster.com/dictionary/proceed one of its meanings is:
- to begin and carry on an action, process, or movement

So in this context it means to start action - begin processing of businessMetrics

Reply:
Hello @korzol 

Do you have any other better naming conventions that we can use instead of proceed? I am reading this code and I do not understand what $businessMetrics->proceed($key) will actually do. What do we do in proceed?

Reply:
Hello @ceckoslab 
Thanks for your review. 
Is it better now?
```
foreach ($this->results as $key => $result) {
    $businessMetrics->generate($key);
    $technicalMetrics->generate($key);
}
```